app/src/main/java/com/artemzin/qualitymatters/network/HostSelectionInterceptor.java

```diff
@@ -0,0 +1,54 @@
+package com.artemzin.qualitymatters.network;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import okhttp3.HttpUrl;
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+
+/**
+ * Such implementation allows us easily change base url in the integration and functional tests!
+ */
+public class HostSelectionInterceptor implements Interceptor {
+
+    @Nullable
+    private volatile String baseUrl;
+
+    @Override @NonNull
+    public Response intercept(@NonNull Chain chain) throws IOException {
+        Request request = chain.request();
+        final String baseUrl = this.baseUrl;
+
+        if (baseUrl != null) {
+            request = modifyRequest(request, baseUrl);
```

Comment:
@artem-zinnatullin, I think that recreating OkHttp is the best option. To achieve that I'm going to use new `@Qualifier` annotation and, consequently, new `@Provides` for new OkHttp instance. Is it OK for you? Do you have another suggestions?